libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/map/NavigationMapboxMap.java

```diff
@@ -165,7 +173,11 @@ public void updateLocation(Location location) {
    * @param maxFpsThreshold to be used to limit map frames per second
    */
   public void updateMapFpsThrottle(int maxFpsThreshold) {
-    mapFpsDelegate.updateMaxFpsThreshold(maxFpsThreshold);
+    if (mapFpsDelegate != null) {
```

Comment:
How about extracting this block of code into a `private` method? You know, we should keep `public` methods as clean as possible so they read like pseudocode (i.e. no conditionals, `for`s, etc.). Same reasoning could be applied in `updateMapFpsThrottleEnabled`, `updateWaynameQueryMap`, `onStart` and `onStop` methods. In general, we should split out the different blocks of related functionality, and in these methods it's pretty noticeable i.e. map way name, map FPS delegate, etc.